packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart

```diff
@@ -0,0 +1,218 @@
+
+  bool fetchInboundRefs = false;
+
+  late ValueNotifier<InboundReferences?> inboundRefsNotifier;
+
+  late void Function(bool) fetchinboundRefsFunc;
+
+  setUp(() {
+    setGlobal(IdeTheme, IdeTheme());
+
+    final json = testInstance.toJson();
+    requestedSize = Instance.parse(json)!;
+    requestedSize.valueAsString = '1024';
+
+    requestFunction = () {
+      requestedSize.valueAsString = size.toString();
+      size++;
+    };
+
+    retainingPathNotifier = ValueNotifier<RetainingPath?>(null);
+
+    fetchRetainingPathFunc = (bool) {
+      if (fetchRetainingPath) {
+        retainingPathNotifier.value = testRetainingPath;
+      } else {
+        fetchRetainingPath = true;
+      }
+    };
+
+    inboundRefsNotifier = ValueNotifier<InboundReferences?>(null);
+
+    fetchinboundRefsFunc = (bool) {
+      if (fetchInboundRefs) {
+        inboundRefsNotifier.value = testInboundRefs;
+      } else {
+        fetchInboundRefs = true;
+      }
+    };
+  });
+
+  testWidgets('test RequestableSizeWidget with null data',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      wrap(
+        RequestableSizeWidget(
+          requestedSize: null,
+          requestFunction: requestFunction,
+        ),
+      ),
+    );
+
+    expect(find.byType(RequestDataButton), findsOneWidget);
+
+    await tester.tap(find.byType(RequestDataButton));
+
+    expect(size, 1);
+  });
+
+  testWidgets('test RequestableSizeWidget with data',
+      (WidgetTester tester) async {
+    size = 100;
+    await tester.pumpWidget(
+      wrap(
+        RequestableSizeWidget(
+          requestedSize: requestedSize,
+          requestFunction: requestFunction,
+        ),
+      ),
+    );
+
+    expect(find.byType(Text), findsOneWidget);
+    expect(find.text('1 KB'), findsOneWidget);
+    expect(find.byType(ToolbarRefresh), findsOneWidget);
+
+    await tester.tap(find.byType(ToolbarRefresh));
+
+    await tester.pumpWidget(
+      wrap(
+        RequestableSizeWidget(
+          requestedSize: requestedSize,
+          requestFunction: requestFunction,
+        ),
+      ),
+    );
+
+    expect(find.text('100 B'), findsOneWidget);
+
+    expect(size, 101);
+  });
+
+  testWidgetsWithWindowSize('test RetainingPathWidget', windowSize,
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      wrap(
+        Column(
+          children: [
+            Flexible(
+              child: RetainingPathWidget(
+                retainingPath: retainingPathNotifier,
+                onExpanded: fetchRetainingPathFunc,
+              ),
+            )
+          ],
+        ),
+      ),
+    );
+
+    expect(find.byType(AreaPaneHeader), findsOneWidget);
+
+    expect(find.text('Retaining Path'), findsOneWidget);
+
+    await tester.tap(find.byType(AreaPaneHeader));
+
+    await tester.pump();
+
+    expect(find.byType(CenteredCircularProgressIndicator), findsOneWidget);
+
+    //collapse then expand to show the retaining path rows
```

Comment:
why do we need to collapse and then expand?

Reply:
It's because the first time expanding, the data is not 'fetched'. It is only testing that the progress indicator is there when the data is null. The second time it is expanded the data is fetched and displayed. Maybe I'll separate the test for when the data is null...